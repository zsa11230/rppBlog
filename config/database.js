"use strict";

/** @type {import('@adonisjs/framework/src/Env')} */
const Env = use("Env");

/** @type {import('@adonisjs/ignitor/src/Helpers')} */
const Helpers = use("Helpers");

module.exports = {
  /*
  |--------------------------------------------------------------------------
  | Default Connection
  |--------------------------------------------------------------------------
  |
  | Connection defines the default connection settings to be used while
  | interacting with SQL databases.
  |
  */
  connection: "mysql",
  /*
  |--------------------------------------------------------------------------
  | MySQL
  |--------------------------------------------------------------------------
  |
  | Here we define connection settings for MySQL database.
  |
  | npm i --save mysql
  |
  */
  mysql: {
    client: "mysql",
    connection: {
      host: Env.get("DB_HOST", "localhost"),
      port: Env.get("DB_PORT", "3306"),
      user: Env.get("DB_USER", "root"),
      password: Env.get("DB_PASSWORD", "root"),
      database: Env.get("DB_DATABASE", "rpp_blog")
    }
  }
};
