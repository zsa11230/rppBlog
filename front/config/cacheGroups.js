exports.cacheGroups = {
  avue: {
    name: 'avue',
    test: /[\\/]node_modules[\\/]@smallwei[\\/]avue[\\/]lib[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  echarts: {
    name: 'echarts',
    test: /[\\/]node_modules[\\/]echarts[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  element: {
    name: 'element',
    test: /[\\/]node_modules[\\/]element-ui[\\/].*/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  antDesignVue: {
    name: 'ant-design-vue',
    test: /[\\/]node_modules[\\/]ant-design-vue[\\/].*/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  antDesign: {
    name: 'ant-design',
    test: /[\\/]node_modules[\\/]@ant-design[\\/].*/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  appChannel: {
    name: 'app-channel',
    test: /[\\/]src[\\/]views[\\/]app[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  // components: {
  //   name: 'components',
  //   test: /[\\/]src[\\/]components[\\/]/,
  //   chunks: 'all',
  //   priority: 1,
  //   reuseExistingChunk: true,
  //   enforce: true
  // },
  wel: {
    name: 'wel',
    test: /[\\/]src[\\/]views[\\/]wel[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  fams: {
    name: 'fams',
    test: /[\\/]src[\\/]views[\\/]fams[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  hrms: {
    name: 'hrms',
    test: /[\\/]src[\\/]views[\\/]hrms[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  exam: {
    name: 'exam',
    test: /[\\/]src[\\/]views[\\/]exam[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  mlms: {
    name: 'mlms',
    test: /[\\/]src[\\/]views[\\/]mlms[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  crms: {
    name: 'crms',
    test: /[\\/]src[\\/]views[\\/]crms[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  gpms: {
    name: 'gpms',
    test: /[\\/]src[\\/]views[\\/]gpms[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  goms: {
    name: 'goms',
    test: /[\\/]src[\\/]views[\\/]goms[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  wenjuan: {
    name: 'wenjuan',
    test: /[\\/]src[\\/]views[\\/]wenjuan[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  admin: {
    name: 'admin',
    test: /[\\/]src[\\/]views[\\/]admin[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  pgbd: {
    name: 'pgbd',
    test: /[\\/]src[\\/]views[\\/]pgbd[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
  govdata: {
    name: 'govdata',
    test: /[\\/]src[\\/]views[\\/]govdata[\\/]/,
    chunks: 'all',
    priority: 1,
    reuseExistingChunk: true,
    enforce: true,
  },
}