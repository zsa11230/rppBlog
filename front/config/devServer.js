
// IP PORT 配置
const baseUrl = process.env.BASEURL
// 设置代理
// netsh interface portproxy add v4tov4 listenport=9999 connectaddress=192.168.1.154 connectport=9999

// 显示所有代理
// netsh interface portproxy show all

// 删除配置: 本机的监听端口为10022,10022端口接受的连接地址为"*",使用的协议为tcp,当前仅支持TCP协议。
// netsh interface portproxy delete v4tov4 listenport=9999 listenaddress=* protocol=tcp
// console.log(process.env)
if (process.env.NODE_ENV === 'development') {
  // 实际代理转发表
  console.log(`Develop Proxy forwarding table, Proxy address: <${baseUrl}>`)
  exports.proxy = {
    '/api': {
      target: baseUrl,
      changeOrigin: true,
      ws: true,
      pathRewrite: {
        '^/api': '',
      },
    },
  }
}
// 改成本地 IP
exports.host = process.env.HOST

exports.port = process.env.PORT
