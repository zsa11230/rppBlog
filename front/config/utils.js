const execSync = require('child_process').execSync
const pack = require('../package.json')

exports.getProject = function () {
  return pack.repository.name.toString().trim()
}

exports.getProjectTeam = function () {
  return pack.repository.team.toString().trim()
}

exports.getProjectName = function () {
  return pack.name.toString().trim()
}

exports.getCurrentTag = function () {
  return pack.version.toString().trim()
}

exports.getGitHash = function () {
  return execSync('git rev-parse HEAD')
    .toString()
    .trim()
}

exports.getProjectDesc = function () {
  const desc = `更新日志为<a href="${pack.repository.changelogurl}">${pack.repository.changelogurl}</a>`
  return desc.toString().trim()
}
