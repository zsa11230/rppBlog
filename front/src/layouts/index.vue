<template>
  <el-container>
    <vec-aside></vec-aside>
    <section class="container" :style="`width: ${containerWidth}`">
      <vec-header></vec-header>
      <el-main>
        <router-view />
      </el-main>
      <vec-footer></vec-footer>
    </section>
  </el-container>
</template>
<script>
import { mapGetters } from 'vuex'
import VecAside from './Components/Aside/index'
import VecHeader from './Components/Header/index'
import VecFooter from './Components/Footer/index'
export default {
  components: {
    VecAside,
    VecHeader,
    VecFooter,
  },
  computed: {
    ...mapGetters(['siderbar_collapsed']),
    containerWidth () {
      return this.siderbar_collapsed ? 'calc(100vw - 64px)' : 'calc(100vw - 200px)'
    },
  },
}
</script>

<style lang="scss" scoped>
.container {
  background-color: #eee;
}
.container ::v-deep {
  .el-main {
    margin: 20px;
    background-color: #fff;
  }
}
</style>
