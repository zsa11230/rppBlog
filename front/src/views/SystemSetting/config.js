export const menuData = [
  {
    id: 1, //id唯一ID标识符
    name: '系统设置',
    path: '/system_setting',
    icon: 'el-icon-s-tools',
    component: 'Layout',
    children: [
      {
        id: 2, //id唯一ID标识符
        name: '用户管理',
        path: 'user',
        icon: '',
        component: 'SystemSetting/User/index',
        children: [],
      },
      {
        id: 3, //id唯一ID标识符
        name: '菜单管理',
        path: 'menu',
        icon: '',
        component: 'SystemSetting/Menu/index',
        children: [],
      },
      {
        id: 4, //id唯一ID标识符
        name: '权限管理',
        path: 'role',
        icon: '',
        component: 'SystemSetting/Role/index',
        children: [],
      },
    ],
  },
]
