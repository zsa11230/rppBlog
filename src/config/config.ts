export default {
  jwtSecret: "BMC",
  token:
    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoicm9vdCIsInBhc3N3b3JkIjoiMTIzNDU2IiwiaWF0IjoxNTY3MzkzNzI1LCJleHAiOjE1Njc0MDgxMjV9.dm0i62VNk0nbuw008l-SjDmcqeEyBNb3oa12UkSh9pM"
};
