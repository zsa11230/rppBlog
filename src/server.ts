import app from "./app/app";
import * as Koa from "koa";
import databaseConnection from "./database/database.connection";

// Process.env will always be comprised of strings, so we typecast the port to a
// number.
const PORT: number = Number(process.env.PORT) || 3000;

function startServe(): Promise<Koa<any, {}>> {
  return new Promise((resolve, reject) => {
    return databaseConnection
      .then(() => {
        if (process.env.NODE_ENV !== "test") {
          app.listen(PORT, () => {
            console.log(`Server running on port http://localhost:${PORT}`);
          });
        }
        resolve(app);
      })
      .catch(err => {
        reject(err);
      });
  });
}

export const eApp = startServe();
