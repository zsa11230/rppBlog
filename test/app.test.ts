import request from "supertest";
import { eApp } from "../src/server";
import Result from "../src/vo/result.vo";
import config from "../src/config/config";
// https://www.rithmschool.com/courses/intermediate-node-express/api-tests-with-jest
// https://molunerfinn.com/Use-Jest-To-Test-Vue-Koa/#Koa%E5%90%8E%E7%AB%AFApi%E6%B5%8B%E8%AF%95
test("Hello koa-typescript-cli", async () => {
  const app = await eApp;
  const response: request.Response = await request(app.callback())
    .get("/")
    .set("Authorization", "Bearer " + config.token);
  expect(response.status).toBe(200);
  const result: Result<string> = JSON.parse(response.text);
  expect(result.data).toBe("hello world koa-typescript-cli");

  expect(response.text).toMatchSnapshot();
});
