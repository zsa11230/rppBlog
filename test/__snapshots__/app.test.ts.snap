// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello koa-typescript-cli 1`] = `"{\\"code\\":0,\\"msg\\":\\"success\\",\\"data\\":\\"hello world koa-typescript-cli\\"}"`;
